components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authcontext/JWTTokenGenerator.java

```diff
@@ -344,25 +317,10 @@ protected SignedJWT signJWTWithRSA(SignedJWT signedJWT, JWSAlgorithm jwsAlgorith
      * @return
      * @throws IdentityOAuth2Exception
      */
+    @Deprecated
     protected JWT signJWT(SignedJWT signedJWT, String tenantDomain, int tenantId)
             throws IdentityOAuth2Exception {
-
-        if (JWSAlgorithm.RS256.equals(signatureAlgorithm) || JWSAlgorithm.RS384.equals(signatureAlgorithm) ||
-                JWSAlgorithm.RS512.equals(signatureAlgorithm)) {
-            return signJWTWithRSA(signedJWT, signatureAlgorithm, tenantDomain, tenantId);
-        } else if (JWSAlgorithm.HS256.equals(signatureAlgorithm) ||
-                JWSAlgorithm.HS384.equals(signatureAlgorithm) ||
-                JWSAlgorithm.HS512.equals(signatureAlgorithm)) {
-            // return signWithHMAC(payLoad,jwsAlgorithm,tenantDomain,tenantId); implementation
-            // need to be done
-        } else if (JWSAlgorithm.ES256.equals(signatureAlgorithm) ||
-                JWSAlgorithm.ES384.equals(signatureAlgorithm) ||
-                JWSAlgorithm.ES512.equals(signatureAlgorithm)) {
-            // return signWithEC(payLoad,jwsAlgorithm,tenantDomain,tenantId); implementation
-            // need to be done
-        }
-        log.error("UnSupported Signature Algorithm");
-        throw new IdentityOAuth2Exception("UnSupported Signature Algorithm");
+        return OAuth2Util.signJWT(signedJWT, signatureAlgorithm, tenantDomain,tenantId);
```

Comment:
Why don't we directly return OAuth2Util.signJWTWithRSA(signedJWT,tenantDomain,tenantId);

Reply:
We can't directly call "signJWTWithRSA", since it will be called based on the signature algorithm we are passing which is checked in "signJWT" method.